packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -669,37 +669,44 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
   }
 
   private createArtifactConfig(props: CanaryProps): CfnCanary.ArtifactConfigProperty | undefined {
-    if (!props.artifactS3EncryptionMode) {
+    if (!props.artifactS3EncryptionMode && !props.artifactS3KmsKey) {
       return undefined;
     }
 
     const isNodeRuntime = props.runtime.family === RuntimeFamily.NODEJS;
-    const isArtifactS3EncryptionModeDefined = !cdk.Token.isUnresolved(props.artifactS3EncryptionMode) && props.artifactS3EncryptionMode;
-    const isArtifactS3KmsKeyDefined = !cdk.Token.isUnresolved(props.artifactS3KmsKey) && props.artifactS3KmsKey;
 
     if (
-      isArtifactS3EncryptionModeDefined &&
-      props.artifactS3EncryptionMode !== ArtifactsEncryptionMode.KMS &&
-      isArtifactS3KmsKeyDefined
+      props.artifactS3EncryptionMode === ArtifactsEncryptionMode.S3_MANAGED &&
+      props.artifactS3KmsKey
```

Comment:
I can see we're no longer checking if these props are tokens. Is it possible for users to provide them as tokens? I don't think it would change much of the logic here, but if they can be tokens, we ought to have unit tests for them.

Reply:
As I understand it, these are not expected to become tokens. Therefore, the token check is unnecessary, so I removed it.